src/nonstandard.jl

```diff
@@ -35,6 +35,7 @@ end
 function tridiagonal_constructor(dl::V, d::V, du::V, du2::V) where {V<:AbstractVector{T}} where T
     Tridiagonal{T,V}(dl, d, du, du2)
 end
+getfields(o::Tridiagonal) = Tuple(getproperties(o))
```

Comment:
Shouldn't we swap these? `getfields` depending on `getproperties` is a bit strange. I'm also not clear why we lose the field names in `getfields`?

Reply:
So you think `getfields` should return a `NamedTuple` (with the exception of `getfields(::Tuple)::Tuple`)? I like that idea, we could then have `getproperties(obj) = getfields(obj)` as a default implementation.